sdk/src/assertions/bmff_hash.rs

```diff
@@ -1208,11 +1230,310 @@ impl BmffHash {
+
+        // build the Merkle tree
+        let m_tree = self.create_merkle_tree_for_merkle_map(reader, box_info, merkle_map)?;
+        let num_leaves = m_tree.leaves.len();
+
+        // save desired Merkle tree row
+        let tree_row = std::cmp::min(max_proofs, m_tree.layers.len() - 1);
+        let merkle_row = m_tree.layers[tree_row].clone();
+        let mut new_hash_row = Vec::new();
+        for mn in merkle_row {
+            let bb = ByteBuf::from(mn.0);
+            new_hash_row.push(bb);
+        }
+        merkle_map.hashes = VecByteBuf(new_hash_row);
+        merkle_map.count = m_tree.leaves.len(); // number of leaves
+
+        // if we are storing the leaves there is no need to generate UUID boxes
+        if tree_row == 0 {
+            return Ok(Vec::new());
+        }
+
+        // generate a Vec containing the BmffMerkleMaps with the proof hashes
+        let mut bmff_merkle_maps = Vec::new();
+        for i in 0..num_leaves {
+            let mut bmff_mm = BmffMerkleMap {
+                unique_id: merkle_map.unique_id,
+                local_id: merkle_map.local_id,
+                location: i,
+                hashes: None,
+            };
+
+            // get proof for this location and replace temp proof
+            let proof = m_tree.get_proof_by_index(bmff_mm.location, max_proofs)?;
+            if !proof.is_empty() {
+                let mut proof_vec = Vec::new();
+                for v in proof {
+                    let bb = ByteBuf::from(v);
+                    proof_vec.push(bb);
+                }
+
+                bmff_mm.hashes = Some(VecByteBuf(proof_vec));
+            }
+
+            bmff_merkle_maps.push(bmff_mm);
+        }
+
+        // generate UUID C2PA Merkle boxes containing each BmffMerkleMap
+        let mut uuid_boxes = Vec::new();
+        for bmff_mm in bmff_merkle_maps {
+            let mm_cbor = serde_cbor::to_vec(&bmff_mm)
+                .map_err(|err| Error::AssertionEncoding(err.to_string()))?;
+
+            uuid_boxes.push(mm_cbor);
+        }
+
+        // pad the uuid boxes  with zeros to the same length as the longest box per the spec
+        let max_length = uuid_boxes.iter().map(|b| b.len()).max().unwrap_or(0);
+        for uuid_box in &mut uuid_boxes {
+            if uuid_box.len() < max_length {
+                uuid_box.resize(max_length, 0);
+            }
+        }
+
+        // convert the cbor Vecs to BMFF UUID boxes
+        uuid_boxes = uuid_boxes
+            .iter()
+            .map(|cbor_data| {
+                let mut uuid_box_data: Vec<u8> = Vec::with_capacity(cbor_data.len() + 32);
+                crate::asset_handlers::bmff_io::write_c2pa_box(
+                    &mut uuid_box_data,
+                    &[],
+                    false,
+                    cbor_data,
+                )
+                .unwrap_or_default(); // this cannot fail unless the vec allocation above fails
```

Comment:
`unwrap_or_default()` is returning `()` either way, should be equivalent when removing this line.